src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -501,20 +515,26 @@ void CliffordSynthesizer::depthHeuristicSynthesis() {
     }
 
     // launch threads
+    std::cout << "Launch threads" << std::endl;
     subCircuits.emplace_back(
         std::async(std::launch::async | std::launch::deferred,
                    [this, startIdx, endIdx, &optimalConfig]() {
                      return cs::CliffordSynthesizer::synthesizeSubcircuit(
                          initialCircuit, startIdx, endIdx, optimalConfig);
                    }));
   }
-
+  std::cout << "Combine Circuits" << std::endl;
   for (auto& subCircuit : subCircuits) {
+    std::cout << "I am still here" << std::endl;
     const auto& circ = subCircuit.get();
+    std::cout << "I am still on" << std::endl;
```

Comment:
According to the [test](https://github.com/cda-tum/mqt-qmap/actions/runs/9429749469/job/25976471788), this line is not reached anymore but the line 528 is.